docs/tutorials/embassy/index.md

```diff
@@ -0,0 +1,148 @@
+slug: /tutorials/embassy
+description: How to install the prerequisites for embassy
+---
+
+# Laboratory Setup
+
+Here, we will cover the steps needed in order to be able to compile and flash Rust applications for `RP2040`, the MCU (Microcontroller Unit) found in our **Raspberry Pi Pico W**s.
+
+## Prerequisites
+
+### Rust Toolchain
+
+In order to install the tools needed to compile Rust code, follow the next steps, depending on your operating system.
+
+#### Linux
+
+Run the this command in terminal:
+
+```shell
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+This downloads and runs `rustup-init.sh`, which in turn downloads and runs the correct version of the `rustup-init` executable for your platform.
+
+#### Windows
+
+Download the respective executable:
+
+* [RUSTUP-INIT.exe - 64bit](https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe)
+* [RUSTUP-INIT.exe - 32bit](https://static.rust-lang.org/rustup/dist/i686-pc-windows-msvc/rustup-init.exe)
+
+> Note: You may be prompted to install [Visual Studio C++ Build tools](https://visualstudio.microsoft.com/visual-cpp-build-tools/). If so, follow the instructions from the previous link.
+
+The last step is to run `rustup --version` in terminal. If everything went well, you should see an output similar to this:
+
+```shell
+rustup 1.26.0 (5af9b9484 2023-04-05)
+info: This is the version for the rustup toolchain manager, not the rustc compiler.
+info: The currently active `rustc` version is `rustc 1.76.0 (07dca489a 2024-02-04)`
+```
+
+### `elf2uf2-rs`
+
+This tool is needed to be able to program the board over `USB`. In order to install it, run the following in your terminal:
+
+```shell
+cargo install elf2uf2-rs
+```
+
+Then, run `elf2uf2-rs --help`. If it was correctly installed, you should see something similar to this in your terminal:
+
+```shell
+Usage: elf2uf2-rs [OPTIONS] <INPUT> [OUTPUT]
+
+Arguments:
+  <INPUT>   Input file
+  [OUTPUT]  Output file
+
+Options:
+  -v, --verbose  Verbose
+  -d, --deploy   Deploy to any connected pico
+  -s, --serial   Connect to serial after deploy
+  -h, --help     Print help
+```
+
+### `probe-rs`
+
+This tool is an embedded debugging and target interaction toolkit. It enables its user to program and debug microcontrollers via a debug probe.
+
+```shell
+cargo install probe-rs --features cli --locked
+```
+
+If you are on **Linux** you will also need to add this [`udev`](https://probe.rs/files/69-probe-rs.rules) file in `/etc/udev/rules.d`. Then, run:
+
+```shell
+udevadm control --reload # to ensure the new rules are used.
+
+udevadm trigger # to ensure the new rules are applied to already added devices.
+```
+
+### VSCode Extension (Optional)
+
+For a better experience, go ahead and install the **Debugger for probe-rs** extension in the Microsoft Extension Marketplace. This will make debugging the program running on the `RP2040` as easy as debugging a Rust program running on your host machine.
```

Comment:
```suggestion
For a better experience, go ahead and install the **Debugger for probe-rs** extension in the Microsoft Extension Marketplace. This will make debugging the program running on the RP2040 as easy as debugging a Rust program running on your host machine.
```